### ISSUE TITLE
Set joystick speed

### ISSUE THREAD

**Issue by qigongFDA** [DIDSR/eeDAP#73]
Current joystick speed is too fast to point on small feature (corner of a cell). We need this to do registration between camera and eyepiece. We should find a method to set joystick speed.

**Comment by qigongFDA** [comment]
There is a method to set the Joystick speed as following. The big disadvantage for this method is user have to disconnect the internet and restart controller. It will be much convenient, if we can set the speed by command line.
1.) Turn off the controller and close any software that is communicating with the controller.
2.) Disconnect any USB cable and RS232 cable from the back of the controller.
3.) Connect the one end of the supplied red X/O ethernet cable to the controller ethernet port and the other end to the PC ethernet port.
4.) Turn on the controller and wait 10 seconds
5.) Open any web browser and navigate to http://192.168.100.1
6.) Click on the "Device Configure" tab.
7.) Click the "CFG" button for Dev Num 1 (X)
8.)  Use the Joystick Speed Slider to adjust the joystick speed.
9.)  Click the "Set Joystick Speed" button
10.) Click the "Apply/Save" button
11.) Repeat steps 6-10 for Dev Num 2 (Y)
12.) Close the browser and turn off the controller.
13.) The settings will take effect when the controller is turned on.

**Comment by qigongFDA** [comment]
There are some commands about Joystick speed: (Top Speed = hold button speed)
1. Get the X-Axis Joystick Normal Speed:
CAN 1 84 18 0 CR
2. Get the Y-Axis Joystick Normal Speed 
CAN 2 84 18 0 CR
3. Set the X-Axis Joystick Normal Speed to 57000 steps/sec 
CAN 1 83 18 57000 CR
4. Set the Y-Axis Joystick Normal Speed to 57000 steps/sec 
CAN 2 83 18 57000 CR
5. Get the X-Axis Joystick Top Speed  
CAN 1 84 19 0 CR
6. Get the Y-Axis Joystick Top Speed  
CAN 2 84 19 0 CR
7. Set the X-Axis Joystick Top Speed to 57000 steps/sec 
CAN 1 83 19 57000 CR
8. Set the Y-Axis Joystick Top Speed to 57000 steps/sec 
CAN 2 83 19 57000 CR

**Comment by brandon-gallas** [comment]
Does this mean that we can set the joystick speed on the fly?
Good progress.

**Comment by qigongFDA** [comment]
Yes. :). I asked three new users to test the 'good speed' for pointing tiny feature. I am adding a new button in "eyepiece and camera registration" window.

**Comment by qigongFDA** [comment]
The above commands change joystick speed for current study. When we restart the stage, the speed goes back to default speed.
For MAC 6000
default normal speed: 3900 steps/sec
default top speed: 280000 steps/sec

**Comment by qigongFDA** [comment]
Add a button in "eyepiece and camera registration" window to set joystick speed. 
The button can show text "Slow Down Joystick" (click to slowdown speed to 2000 steps/sec) or "Recover Joystick Speed" (click to recover speed to 3900 steps/sec) . 
After eyepiece and camera registration, the joystick will recover to default speed (3900 steps/sec).
peed to 3900 steps/sec) . 
After eyepiece and camera registration, the joystick will recover to default speed (3900 steps/sec).
